### PR TITLE
Added aditional routes and Event Schema changed!

### PR DESCRIPTION
**Pull request by HristoYonkov**
At       "http://localhost:3030/events/:id" - youcan access exact Event object from data!.
At       "http://localhost:3030/events/" - you can access all Event objects from data.
added additional validation on properties on Event model, also "likes" and "creator" properties was changed to accept object id's of the User model or  Organization model.

### CODE REVIEW

File: src/models/Event.js [hunk 1]
```diff
@@ -39,27 +40,38 @@ const eventSchema = new Schema({
...
-            endTime: { type: String }
         }]
     },
-    imageUrl: { type: String, required: true },
+    imageUrl: {
+        type: String,
+        required: true,
+        validate: {
+            validator: (value) => validString.test(value),
+            message: "Invalid URL, must start with HTTP:///HTTPS://",
```
Comment (Getsov) [inline src/models/Event.js:51]: Won’t the organisation be able to upload an image, not only passing an url for it?

File: src/models/Event.js [hunk 2]
```diff
@@ -39,27 +40,38 @@ const eventSchema = new Schema({
...
+        },
+    },
     contacts: {
         // TODO: Check later for unique COORDS..!
-        coordinates: { lat: { type: String }, long: { type: String } },
+        coordinates: { lat: { type: String, required: true }, long: { type: String, required: true } },
```
Comment (Getsov) [inline src/models/Event.js:56]: Nobody’s going to know the event’s place coordinates. We must convert them, they are not going to come as an input

File: src/models/Event.js [hunk 3]
```diff
@@ -39,27 +40,38 @@ const eventSchema = new Schema({
...
-    creator: { type: ObjectId, ref: 'User', required: true }, // User who created the event (you can specify user properties here)
-    winners: [{ name: { type: String } }, { name: { type: String } }, { name: { type: String } }], // Array of event winners (you can specify winner properties here)
+    category: {
+        type: String,
+        required: true,
+        minlength: [2, "Category must be minimum 2 characters long!"],
+        maxlength: [30, "Category must be maximum 30 characters long!"],
+    },
+    likes: [{ type: ObjectId, ref: 'User' } || { type: ObjectId, ref: 'Organization' }],
+    creator: { type: ObjectId, ref: 'User', required: true } || { type: ObjectId, ref: 'Organization', required: true },
```
Comment (Getsov) [inline src/models/Event.js:69]: User cannot be a creator, only admin or organisation

File: src/services/eventsService.js [hunk 4]
```diff
@@ -10,24 +10,44 @@ async function registerEvent() {
             startDate: Date.now(),
             startTime: '23:59',
             endTime: '06:40'
-        }], // Eventually add another date!
+        }],
         imageUrl: 'https://cdn.akamai.steamstatic.com/steam/apps/1839940/capsule_616x353.jpg?t=1683123302',
         contacts: {
-            coordinates: { lat: '', long: '' },
+            coordinates: { lat: 'sda', long: 'sad' },
```
Comment (Getsov) [inline src/services/eventsService.js:16]: These are not valid coordinates, if the front end tries to test visualisation of them, the code will break

File: src/models/Event.js (hunk 3 above)
Comment (HristoYonkov) [reply src/models/Event.js:69]: I will check later if user have admin property in controller function, also I dont have the updated version of user.

File: src/models/Event.js (hunk 2 above)
Comment (HristoYonkov) [reply src/models/Event.js:56]: Sorry i dod not understand you can you rephrase yourself?

File: src/models/Event.js (hunk 3 above)
Comment (HristoYonkov) [reply src/models/Event.js:69]: We have not talk about admin model we talk about user Model with admin property............

**Comment by HristoYonkov** [comment]
Check the changes agaian.

File: src/models/Event.js (hunk 1 above)
Comment (HristoYonkov) [reply src/models/Event.js:50]: Did you mean to store "img" files locally?

File: src/models/Event.js [hunk 5]
```diff
@@ -39,27 +40,51 @@ const eventSchema = new Schema({
...
+        },
+    },
+    imageData: {
+        data: Buffer,
+        connectType: String
+    },
     contacts: {
         // TODO: Check later for unique COORDS..!
-        coordinates: { lat: { type: String }, long: { type: String } },
+        coordinates: {
```
Comment (Getsov) [inline src/models/Event.js:59]: No, the deleted object was correct, without validation, only type string is enough, not needed coordinates in coordinates and type type in coordinates again

File: src/models/Event.js (hunk 3 above)
Comment (Getsov) [reply src/models/Event.js:82]: Which updated version of user? Make TODO’s so we don’t forget it later if you are going to check it later.

File: src/models/Event.js [hunk 6]
```diff
@@ -39,27 +40,51 @@ const eventSchema = new Schema({
...
+        type: String,
+        required: true,
+        minlength: [2, "Category must be minimum 2 characters long!"],
+        maxlength: [30, "Category must be maximum 30 characters long!"],
+    },
+    // TODO: check if the User has admin role.
+    likes: [{ type: ObjectId, ref: 'User' } || { type: ObjectId, ref: 'Organization' }],
+    creator: { type: ObjectId, ref: 'User', required: true } || { type: ObjectId, ref: 'Organization', required: true },
+    winners: [
+        { name: String, vehicle: String },
```
Comment (Getsov) [inline src/models/Event.js:84]: Why 3 objects in the array?

File: src/services/eventsService.js [hunk 7]
```diff
@@ -3,31 +3,51 @@ const Event = require('../models/Event');
...
         dates: [{
             startDate: Date.now(),
             startTime: '23:59',
             endTime: '06:40'
-        }], // Eventually add another date!
+        }],
         imageUrl: 'https://cdn.akamai.steamstatic.com/steam/apps/1839940/capsule_616x353.jpg?t=1683123302',
         contacts: {
-            coordinates: { lat: '', long: '' },
+            coordinates: { coordinates: [42.52911093579847, 24.707900125838766] },
```
Comment (Getsov) [inline src/services/eventsService.js:16]: Correct it based on my comment from the model

File: src/models/Event.js (hunk 6 above)
Comment (HristoYonkov) [reply src/models/Event.js:84]: cuz winner may be will be 3 ?

Comment (Getsov) [reply src/models/Event.js:84]: Cuz?? So likes (two rows above) are going to be only 1 ?

File: src/models/Event.js [hunk 8]
```diff
@@ -68,15 +69,11 @@ const eventSchema = new Schema({
...
+
     likes: [{ type: ObjectId, ref: 'User' } || { type: ObjectId, ref: 'Organization' }],
     // TODO: We must add Admin model later!
     creator: { type: ObjectId, ref: 'Admin', required: true } || { type: ObjectId, ref: 'Organization', required: true },
-    winners: [
-        { name: String, vehicle: String },
-        { name: String, vehicle: String },
-        { name: String, vehicle: String }
-    ],
+    winners: { firstPlace: { name: String, vehicle: String } },
```
Comment (Getsov) [inline src/models/Event.js:76]: No, just keep it an array with objects with properties name, vehicle and place.

File: src/services/eventsService.js [hunk 9]
```diff
@@ -20,13 +20,14 @@ async function registerEvent() {
...
+        likes: ['6542c24b6102c6f4e79108fc', '6542c24b6102c6f4e79108fc', '6542c24b6102c6f4e79108fc'],
         creator: '6542c24b6102c6f4e79108fc',
-        winners: [
-            { name: 'Pavel', vehicle: "Trabant" },
-            { name: 'Pavel', vehicle: "Trabant" },
-            { name: 'Pavel', vehicle: "Trabant" }
-        ],
+        winners: {
+            firstPlace: { name: 'Pavel', vehicle: "Trabant" },
+            sePlace: { name: 'Pavel', vehicle: "Trabant" },
```
Comment (Getsov) [inline src/services/eventsService.js:27]: Update needed

File: src/models/Event.js (hunk 8 above)
Comment (HristoYonkov) [reply src/models/Event.js]: Yes this is good approach, just objects and inside will be the place.

File: src/models/Event.js [hunk 10]
```diff
@@ -69,11 +69,34 @@ const eventSchema = new Schema({
         minlength: [2, "Category must be minimum 2 characters long!"],
         maxlength: [30, "Category must be maximum 30 characters long!"],
     },
-
     likes: [{ type: ObjectId, ref: 'User' } || { type: ObjectId, ref: 'Organization' }],
     // TODO: We must add Admin model later!
     creator: { type: ObjectId, ref: 'Admin', required: true } || { type: ObjectId, ref: 'Organization', required: true },
-    winners: { firstPlace: { name: String, vehicle: String } },
+    winners: {
```
Comment (Getsov) [inline src/models/Event.js:75]: winners: [
 {
  name: { type: String, required: true },
  vehicle: { type: String },
  place: { type: String, required: true }
 }
]

**Review by Getsov — CHANGES_REQUESTED** [review]

**Review by Getsov — APPROVED** [review]
